src/program/config/set_alarm_operator_state/set_alarm_operator_state.lua

```diff
@@ -0,0 +1,42 @@
+-- Use of this source code is governed by the Apache 2.0 license; see COPYING.
+module(..., package.seeall)
+
+local common = require("program.config.common")
+
+local function usage(exit_code)
+   print(require("program.config.set_operator_state.README_inc"))
+   main.exit(exit_code)
+end
+
+local function parse_args (args)
+   local function parse_alarm_key (key)
+      local t = {}
+      for each in key:gmatch("([^,]+)") do
+         table.insert(t, each)
+      end
+      return {resource=t[1], alarm_type_id=t[2], alarm_type_qualifier=t[3] or ''}
```

Comment:
Is this a standard syntax for identifying alarms?  Can you document it somewhere?  Are there any of these arguments that are required (alarm type id?)?

Reply:
Yes, according the ietf-alarms schema and alarm key is composed of three elements: `key`, `alarm-type` and `alarm-type-qualifier`. `alarm-type-qualifier`, unlike `resource` and `alarm-type`, is not defined as mandatory, thus is optional.

```yang
container alarm {
   list alarm {
      key "resource alarm-type-id alarm-type-qualifier";
   }
}

grouping common-alarm-parameters {
   leaf resource {
      type resource;
      mandatory true;
   }
   leaf alarm-type-id {
      type alarm-type-id;
      mandatory true;
   }
   leaf alarm-type-qualifier {
       type alarm-type-qualifier;
       description
                "This leaf is used when the 'alarm-type-id' leaf cannot
                uniquely identify the alarm type.  Normally, this is not
                the case, and this leaf is the empty string.";
   }
}
```

I'm going to update the README to add this info.

Reply:
So it sounds like you need 2 or more arguments.  I don't think the code above ensures this though.  Could we instead separate the resource and alarm-type-id arguments?  If there's a qualifier we can attach it to the alarm type id.  e.g.

```
snabb config set-operator-state lwaftr external-interface arp-resolution ack
```

If there's a qualifier it could be `arp-resolution/foo`.  Dunno.  Alternately if there is a usual textual interface for this then we could use that; comma to me suggests a sequence of like things instead of a hierarchy of selectors.